src/tracer/src/nondaemon_commands.rs

```diff
@@ -221,28 +223,61 @@ pub async fn setup_config(
 
 pub async fn update_tracer() -> Result<()> {
     let octocrab = octocrab::instance();
-
     let release = octocrab
         .repos(REPO_OWNER, REPO_NAME)
         .releases()
         .get_latest()
         .await?;
 
-    if release.tag_name == env!("CARGO_PKG_VERSION") {
-        println!("You are already using the latest version of Tracer.");
+    let current = env!("CARGO_PKG_VERSION");
+    let latest = &release.tag_name;
+
+    let current_ver = Version::from_str(current)?;
+    let latest_ver = Version::from_str(latest)?;
+
+    if latest_ver <= current_ver {
+        println!(
+            "\nTracer is already at the latest version: {}.",
+            current_ver
+        );
+        return Ok(());
+    }
+
+    println!("\nA new version of Tracer is available!");
+    println!("\nVersion Information:");
+    println!("  Current Version: {}", current_ver);
+    println!("  Latest Version:  {}", latest_ver);
+
+    println!("\nWould you like to proceed with the update? [y/N]");
+    let mut input = String::new();
+    std::io::stdin().read_line(&mut input)?;
+
+    if !input.trim().eq_ignore_ascii_case("y") {
+        println!("Update cancelled by user.");
         return Ok(());
     }
 
     let config = Config::default();
 
-    println!("Updating Tracer to version {}", release.tag_name);
+    println!("\nUpdating Tracer to version {}...", latest_ver);
 
     let mut command = Command::new("bash");
-    command.arg("-c").arg(format!("curl -sSL https://raw.githubusercontent.com/davincios/tracer-daemon/main/install-tracer.sh | bash -s -- {} && . ~/.bashrc && tracer", config.api_key));
-
-    command
+    command.arg("-c").arg(format!(
+        "curl -sSL https://install.tracer.cloud | sudo bash -s -- {} && . ~/.bashrc",
```

Comment:
we are not sending anymore the APIKEY the command should be
`curl -sSL https://install.tracer.cloud/ | sudo bash && source ~/.bashrc && source ~/.zshrc`